test/PositionContext.test.ts

```diff
@@ -465,17 +461,41 @@ suite("PositionContext", () => {
         });
     });
 
-    suiteTODODisabled("completionItems", () => {
+    suite("completionItems", () => {
         function completionItemsTest(documentText: string, index: number, expectedCompletionItems: Completion.Item[]): void {
             test(`with ${Utilities.escapeAndQuote(documentText)} at index ${index}`, () => {
                 const dt = new DeploymentTemplate(documentText, "id");
                 const pc: PositionContext = dt.getContextFromDocumentCharacterIndex(index);
                 return pc.completionItems.then((completionItems: Completion.Item[]) => {
-                    assert.deepStrictEqual(completionItems, expectedCompletionItems);
+                    compareTestableCompletionItems(completionItems, expectedCompletionItems);
                 });
             });
         }
 
+        function compareTestableCompletionItems(actualItems: Completion.Item[], expectedItems: Completion.Item[]): void {
+            let isFunctionCompletions = expectedItems.some(item => allTestableCompletionNames.has(item.name));
+
+            // Ignore functions that aren't in our testing list
+            if (isFunctionCompletions) {
+                // Unless it's an empty list - then we want to ensure the actual list is empty, too
+                if (expectedItems.length > 0) {
+                    actualItems = actualItems.filter(item => allTestableCompletionNames.has(item.name));
+                }
+            }
+
+            // Make it easier to see missing names quickly
+            let actualNames = actualItems.map(item => item.name);
+            let expectedNames = expectedItems.map(item => typeof item === 'string' ? item : item.name);
+            assert.deepStrictEqual(actualNames, expectedNames, "Names in the completion items did not match");
+
+            assert.deepStrictEqual(actualItems, expectedItems);
+        }
+
+        // IMPORTANT
+        // We will only test against the completions in this list.  Any new functions added to the ExpressionMetadata.json file
```

Comment:
Is this really "too difficult" because people update "ExpressionMetadata.json" a lot or is it just because the tests weren't being run with CI and have gotten out of date?

If the file is actually updated frequently, then I'm okay with your change. But can we at least print the ones we're ignoring so that someone might fix them later? If there's no value in ever fixing them later - I'm not sure why we even have these tests in the first place

Reply:
They still add new functions, although less commonly than before (there are about twice as many as when these test were written).  Have you looked at the tests?  They are *very* specific on exactly what happens at every single index in every single string.  Which is overtesting.  I do want it to be easy to update the function metadata.  These tests make it hard.

The tests are very good at the general issue of handling Intellisense and are very valuable there. But that functionality should be tested separate of the exact function metadata, which is what I'm doing by restricting it to current functions.  Another possible approach would have been to copy the current data and place it into the test.

So there's no need to "fix" these tests later - this is not a kludge, I'm just restricting them to testing actual logic and not the metadata itself.  I don't think adding tests for the actual metadata would make a lot of sense unless we find things regressing.

Reply:
Added some comments to this effect.